kratos/utilities/variable_utils.h

```diff
@@ -803,44 +875,43 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
     }
 
     /**
-     * @brief Returns the nodal value summation of an historical vector variable.
-     * @param rVar reference to the vector variable to summed
-     * @param rModelPart reference to the model part that contains the objective node set
-     * @return sum_value summation vector result
+     * @brief This method accumulates and return a variable value
+     * For a nodal historical variable, this method accumulates and
+     * returns the summation in a model part.
+     * @tparam TDataType Variable datatype
+     * @tparam Variable<TDataType> Variable type
+     * @param rVariable Nodal historical variable to be accumulated
+     * @param rModelPart Model part in where the summation is done
+     * @param BuffStep Buffer position
+     * @return TDataType Value of the summation
      */
-    array_1d<double, 3> SumHistoricalNodeVectorVariable(
-        const ArrayVarType& rVar,
-        const ModelPart& rModelPart,
-        const unsigned int rBuffStep = 0
-        );
-
-    /**
-     * @brief Returns the nodal value summation of an historical scalar variable.
-     * @param rVar reference to the scalar variable to be summed
-     * @param rModelPart reference to the model part that contains the objective node set
-     * @return sum_value: summation result
-     */
-    template< class TVarType >
-    double SumHistoricalNodeScalarVariable(
-        const TVarType& rVar,
-        const ModelPart& rModelPart,
-        const unsigned int rBuffStep = 0
+    template< class TDataType, class TVarType = Variable<TDataType> >
+    TDataType SumHistoricalVariable(
+        const TVarType &rVariable,
+        const ModelPart &rModelPart,
+        const unsigned int BuffStep = 0
         )
     {
         KRATOS_TRY
 
-        double sum_value = 0.0;
+        TDataType sum_value;
+        AuxiliaryInitializeValue(sum_value);
```

Comment:
I have the impression you could use one of the other functions in the implementation of this one (maybe i am wrong)

Reply:
I think it is even better to use the `Zero()` method of the `Variable` class. Thanks for spotting it.

Reply:
I was wrong, the component variable has no `Zero()` method. I'd leave it as it is (I don't think what you suggest can be done).

Reply:
ok